src/remote/watcher.js

```diff
@@ -34,15 +34,30 @@ export default class RemoteWatcher {
   }
 
   start () {
-    this.watch()
-    this.intervalID = setInterval(this.watch, HEARTBEAT)
+    this.started = new Promise((resolve, reject) => {
+      this.started.resolve = resolve
+      const watch () => {
+        this.watch().catch(err => {
+          reject(err)
```

Comment:
`start()` is slow, but starting the `LocalWatcher` is slow too. And, IIRC, it's better to have synchronized all the metadata before starting to poll the sync queue for applying changes.